### PR TITLE
[FIX] l10n_latam_invoice_document: refund doc type

### PR DESCRIPTION
**Pull request by jjscarafia**
Prior to this change, if the user use a full refund option and choose a document type different than default one, the default document type was use regardless of the choosen one.

This was broken on this commit https://github.com/odoo/odoo/commit/dfd01b8c5c7e1177f37bf199790a0732a61eed78#diff-1369ca152be95632086495cfcfbd54a0d18d893c2389670ff1d516f1596f3e0d

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_latam_invoice_document/wizards/account_move_reversal.py [hunk 1]
```diff
@@ -52,7 +51,8 @@ def _compute_document_type(self):
                     'partner_id': record.move_ids.partner_id.id,
                     'company_id': record.move_ids.company_id.id,
                 })
-                record.l10n_latam_document_type_id = refund.l10n_latam_document_type_id
+                if record.l10n_latam_document_type_id not in refund.l10n_latam_available_document_type_ids._origin:
```
Comment (vbe-odoo) [inline addons/l10n_latam_invoice_document/wizards/account_move_reversal.py:54]: @jjscarafia check this piece

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by jco-odoo** [comment]
@robodoo retry

**Comment by jco-odoo** [comment]
@jjscarafia There are the tests in l10n_ar_reports and in test_l10n_cl_account_move_reversal_partial_refund that fail, because the document type is simply not passed.  Can we also remove https://github.com/odoo/odoo/pull/82670/files#diff-1369ca152be95632086495cfcfbd54a0d18d893c2389670ff1d516f1596f3e0dR53 please?  Why not reuse part of the method in account.move instead of creating something on the fly like that?

**Comment by vbe-odoo** [comment]
@jjscarafia for this one Josse left a comment saying that some tests were failing and were not possible to merge. Can you help me with this one?
Thanks!
cc' @ren-odoo @fvz-odoo @zaoral

**Comment by jjscarafia** [comment]
Hi @jco-odoo I'm taking a look to issue with tests.
Regarding that [new](https://github.com/odoo/odoo/pull/82670/files#diff-1369ca152be95632086495cfcfbd54a0d18d893c2389670ff1d516f1596f3e0dL49) record to get l10n_latam_available_document_type_ids value, I don't see a better and safer way because we don't know which values use each localization to compute the domain. The [method that build the domain](https://github.com/odoo/odoo/blob/master/addons/l10n_latam_invoice_document/models/account_move.py#L172) use an instance of an invoice (account.move) and is inherited by the localizations where each one can use the fields of the invoice that they need to compute it.

**Comment by jjscarafia** [comment]
@jco-odoo 
should be fixed, this was the change
```
-        for record in self.filtered(lambda x: x.l10n_latam_document_type_id not in x.l10n_latam_available_document_type_ids):
-            record.l10n_latam_document_type_id = record.l10n_latam_available_document_type_ids[0] if record.l10n_latam_available_document_type_ids else False
+        for record in self.filtered(
+                lambda x: not x.l10n_latam_document_type_id or
+                x.l10n_latam_document_type_id not in x.l10n_latam_available_document_type_ids):
+            document_types = record.l10n_latam_available_document_type_ids._origin
+            record.l10n_latam_document_type_id = document_types[0] if document_types else False

```

File: addons/l10n_latam_invoice_document/wizards/account_move_reversal.py [hunk 2]
```diff
@@ -32,10 +32,17 @@ def _reverse_type_map(self, move_type):
             'in_receipt': 'out_receipt'}
         return match.get(move_type)
 
-    @api.depends('move_ids')
+    @api.depends('l10n_latam_available_document_type_ids')
     def _compute_document_type(self):
+        for record in self.filtered(
+                lambda x: not x.l10n_latam_document_type_id or
+                x.l10n_latam_document_type_id not in x.l10n_latam_available_document_type_ids):
+            document_types = record.l10n_latam_available_document_type_ids._origin
```
Comment (jco-odoo) [inline addons/l10n_latam_invoice_document/wizards/account_move_reversal.py:40]: @jjscarafia Why do you do ._origin here?

Comment (vbe-odoo) [reply addons/l10n_latam_invoice_document/wizards/account_move_reversal.py:40]: @zaoral if time allows please check this

Comment (Bruno-Zanotti) [reply addons/l10n_latam_invoice_document/wizards/account_move_reversal.py:40]: @jjscarafia @jco-odoo @zaoral @vbe-odoo 
I think the reason is that the "_origin" was already on the original method.. so we should check if we can remove it from there too.. this is the [line](https://github.com/odoo/odoo/blob/19a2f84b39779f7bb68804246bfe28d106da812d/addons/l10n_latam_invoice_document/wizards/account_move_reversal.py#L46)

Comment (jjscarafia) [reply addons/l10n_latam_invoice_document/wizards/account_move_reversal.py:40]: Without the ._origin it doesn't work properly. I need to added to make it work. If you remove it and play with it you will find the error (if not, I can make it on my return).

I remember wi discussed the original "._origin" and was something related to a limitartion / bug of the ORM that on m2m computed fields was not working as other fields (._origin solve it)

@Bruno-Zanotti @zaoral @jco-odoo

Comment (vbe-odoo) [reply addons/l10n_latam_invoice_document/wizards/account_move_reversal.py:40]: @jco-odoo could you check this last comment from @jjscarafia about the "._origin", thanks!

Comment (vbe-odoo) [reply addons/l10n_latam_invoice_document/wizards/account_move_reversal.py:40]: @jjscarafia if you can share the use case (example or screenshots) w/the issue will be nice to have it

**Comment by jjscarafia** [comment]
I've sent it on the tax, but to add it here, basically there are two issues related to why we need ._origin (or .ids).
The m2m computed field l10n_latam_available_document_type_ids NewId recordsets and doing that:
1. this condition don't work `x.l10n_latam_document_type_id not in x.l10n_latam_available_document_type_ids`
2. neither assigning the NewId to `record.l10n_latam_document_type_id`

**Comment by ren-odoo** [comment]
Hi @jco-odoo  can you help us to take a look at the last JJ's comment please?

Thank you.

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**